crates/common/tedge_config/src/sudo.rs

```diff
@@ -0,0 +1,53 @@
+use std::ffi::OsStr;
+use std::path::Path;
+use std::process::Command;
+use std::sync::Arc;
+
+use tracing::warn;
+
+use crate::TEdgeConfig;
+
+const SUDO: &str = "sudo";
+
+/// An object used to remember the user's sudo preference.
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct SudoCommandBuilder {
+    path: Option<Arc<Path>>,
+    enabled: bool,
+}
```

Comment:
I would remove the `enable` field, as it's redundant with the `None` case.

```suggestion
#[derive(Debug, PartialEq, Eq, Clone)]
pub struct SudoCommandBuilder {
    path: Option<Arc<Path>>,
}
```